lib/cast.js

```diff
@@ -226,12 +226,9 @@ module.exports = function cast(schema, obj, options) {
           }
         }
       } else if (val.constructor.name === 'Array' && ['Buffer', 'Array'].indexOf(schematype.instance) === -1) {
-        var casted = [];
-        for (var valIndex = 0; valIndex < val.length; valIndex++) {
-          casted.push(schematype.castForQuery(val[valIndex]));
-        }
+        var castedValues = val.map(schematype.castForQuery.bind(schematype));
```

Comment:
Can you revert this change? `map()` is very neat and clean but much slower than for loop, and `.bind()` is extra slow.